src/main/java/salted/calmmornings/common/utils/DespawnUtils.java

```diff
@@ -48,24 +49,36 @@ public static void despawnEntities(Level level, ServerPlayer player) {
         }
     }
 
+    private static boolean isCategoryEnabled(String category) {
+        return switch (category) {
+            case "CREATURE" -> Config.CREATURE.get();
+            case "MONSTER" -> Config.MONSTER.get();
+            case "AMBIENT" -> Config.AMBIENT.get();
+            case "AXOLOTLS" -> Config.AXOLOTLS.get();
+            case "WATER_CREATURE" -> Config.WATER_CREATURE.get();
+            case "UNDERGROUND_WATER_CREATURE" -> Config.UNDERGROUND_WATER_CREATURE.get();
+            case "WATER_AMBIENT" -> Config.WATER_AMBIENT.get();
+            case "MISC" -> Config.MISC.get();
+            default -> false;
+        };
+    }
+
     // private methods for despawning entities
-    private static boolean shouldDespawn(@NotNull Entity entity) {
-        EntityType<?> type = entity.getType();
-        String entityKey = EntityType.getKey(type).toString();
-
-        // separate mod/entity ids
-        String[] key = entityKey.split(":");
-        String modId = key[0];
-        String entityId = key[1];
-
-        // get mobCategory and categoryFilter for check
-        ConcurrentHashMap<String, ConcurrentHashMap<String, ListInfo>> map = ListBuilder.getEntityMap();
-        ListInfo listInfo = map.get(modId).get(entityId);
-        MobCategory mobCategory = listInfo.getCategory();
-        HashSet<MobCategory> categoryFilter = ListBuilder.getFilterList();
-
-        if (Config.ENABLE_LIST.get()) return listInfo.getDespawnable() && categoryFilter.contains(mobCategory);
-        return (listInfo.getCategory() == MobCategory.MONSTER && listInfo.getDespawnable());
+    private static boolean shouldDespawn(@NotNull Entity e) {
+        if (!Config.ENABLE_LIST.get()) return shouldDespawnBuiltin(e);
+        ResourceLocation key = EntityType.getKey(e.getType());
+        boolean match = Config.MOB_LIST.get().contains(key.toString())  || Config.MOB_LIST.get().contains(key.getNamespace() + ":*");
+
+        String category = Config.CATEGORY.getOrDefault(key.toString(), e.getType().getCategory().toString());
```

Comment:
An adjustment needs to be made. This line and the line underneath it should be swapped. It should come out looking like:
`String category = Config.CATEGORY.getOrDefault(key.getNamespace() + ":*", e.getType().getCategory().toString());
        category = Config.CATEGORY.getOrDefault(key.toString(), category)`